hs_core/views/__init__.py

```diff
@@ -339,6 +339,31 @@ def delete_resource(request, shortkey, *args, **kwargs):
 
     return HttpResponseRedirect('/my-resources/')
 
+
+def rep_res_bag_to_irods_user_zone(request, shortkey, *args, **kwargs):
+    if request.is_ajax:
+        res, authorized, user = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
+        if not authorized:
```

Comment:
```authorize()``` function will raise exception if the authorization fails. That means the ```if not authorized``` will never be executed. Alternatively you can set the parameter ```raise_exception=False``` when calling ```authorize()```